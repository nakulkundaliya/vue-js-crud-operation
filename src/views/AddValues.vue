<template>
  <div>
    <CRow>
      <CCol sm="6">
        <CCard>
          <CForm @submit.prevent="add">
            <CCardHeader>
              <strong>Add Values </strong><small>Form</small>
            </CCardHeader>
            <CCardBody>
              <CTextarea
                label="Name"
                placeholder="Enter your values "
                v-model="value"
              />
            </CCardBody>
            <CCardFooter>
              <CButton type="submit" size="sm" color="primary"
                ><CIcon name="cil-check-circle" /> Save</CButton
              >
            </CCardFooter>
          </CForm>
        </CCard>
      </CCol>
    </CRow>
  </div>
</template>

<script>
import { apiMethods } from "../state/helpers";

export default {
  name: "AddValues",
  methods: {
    ...apiMethods,
    add() {
      this.isLoading = true;
      // Reset the authError if it existed.
      this.error = null;
      return this.addValue({
        value: this.value,
      })
        .then(() => {
          this.isLoading = false;
          // Redirect to the originally requested page, or to the home page
          this.$router.back();
        })
        .catch((error) => {
          this.isLoading = false;
          this.error = error;
        });
    },
    validator(val) {
      return val ? val.length >= 4 : false;
    },
  },
};
</script>
