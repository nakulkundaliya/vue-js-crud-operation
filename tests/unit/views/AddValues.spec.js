import Vue from 'vue';
import { shallowMount } from '@vue/test-utils';
import CoreuiVue from '@coreui/vue';
import AddValues from '@/views/AddValues';

Vue.use(CoreuiVue);

describe('AddValues.vue', () => {
  it('has a name', () => {
    expect(AddValues.name).toBe('AddValues');
  });
  it('is Vue instance', () => {
    const wrapper = shallowMount(AddValues);
    expect(wrapper.vm).toBeTruthy();
  });
  it('is AddValues', () => {
    const wrapper = shallowMount(AddValues);
    expect(wrapper.findComponent(AddValues)).toBeTruthy();
  });

  test('renders correctly', () => {
    const wrapper = shallowMount(AddValues);
    AddValues;
    expect(wrapper.element).toMatchSnapshot();
  });
});
