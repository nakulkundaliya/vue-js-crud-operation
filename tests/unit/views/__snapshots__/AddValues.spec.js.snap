// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddValues.vue renders correctly 1`] = `
<div>
  <crow-stub
    gutters="true"
    tag="div"
  >
    <ccol-stub
      sm="6"
      tag="div"
    >
      <ccard-stub>
        <cform-stub>
          <ccardheader-stub>
            <strong>
              Add Values 
            </strong>
            <small>
              Form
            </small>
          </ccardheader-stub>
           
          <ccardbody-stub>
            <ctextarea-stub
              label="Name"
              lazy="400"
              placeholder="Enter your values "
            />
          </ccardbody-stub>
           
          <ccardfooter-stub>
            <cbutton-stub
              activeclass="router-link-active"
              color="primary"
              event="click"
              exactactiveclass="router-link-exact-active"
              routertag="a"
              size="sm"
              target="_self"
              type="submit"
            >
              <cicon-stub
                name="cil-check-circle"
              />
               Save
            </cbutton-stub>
          </ccardfooter-stub>
        </cform-stub>
      </ccard-stub>
    </ccol-stub>
  </crow-stub>
</div>
`;
